risc0/zkvm/sdk/rust/host/src/lib.rs

```diff
@@ -322,7 +359,21 @@ mod test {
         prover.add_input(vec.as_slice()).unwrap();
         let receipt = prover.run()?;
         receipt.verify(IO_ID).unwrap();
-        Ok(())
+        Ok(receipt)
+    }
+
+    #[test]
+    fn receipt_serde() {
+        // TODO(nils): Move this constant into something both the guest and host can
+        // depend on
+        const HEAP_START: u32 = 0x0008_0000;
+
+        let receipt: Receipt = run_memio(&[(HEAP_START, 0)]).unwrap();
+        let ser: Vec<u32> = risc0_zkvm_serde::to_vec(&receipt).unwrap();
+        let de: Receipt = risc0_zkvm_serde::from_slice(&ser).unwrap();
+        assert_eq!(de.get_journal().unwrap(), receipt.get_journal().unwrap());
+        assert_eq!(de.get_seal().unwrap(), receipt.get_seal().unwrap());
+        de.verify(IO_ID).unwrap();
```

Comment:
I think you also want to assert that the verify succeeded?

Reply:
The C++ API throws an error when the verify fails and does not return a bool.  When we lift the rust verifier into zkvm/sdk/rust/host we will need to check this, though.